### PR TITLE
SWARM-768: Honor swarm.project.stage as an environment variable

### PR DESCRIPTION
**Pull request by gastaldi**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

---
## Motivation

Docker environments use env vars to pass parameters to docker container
## Modifications

SwarmProperties.PROJECT_STAGE is resolved in System.getenv() if System.getProperty() doesn't return null
## Result

Swarm uses SwarmProperties.PROJECT_STAGE

### CODE REVIEW

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java [hunk 1]
```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
 
     private void loadStageConfiguration(URL url) {
         List<ProjectStage> projectStages = new ProjectStageFactory().loadStages(url);
-        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE, "default");
+        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE);
```
Comment (lordofthejars) [inline core/container/src/main/java/org/wildfly/swarm/Swarm.java:467]: I like the if/else approach but it seems that in Java 8 it is better to use Optionals.

```
Optional.ofNullable(System.getenv(PROJECT_STAGE))
                .orElse(Optional.ofNullable(System.getProperty(PROJECT_STAGE))
                .orElse("default"));
```

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java [hunk 2]
```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
...
+        if (stageName == null) {
+            stageName = System.getenv(SwarmProperties.PROJECT_STAGE);
```
Comment (lordofthejars) [inline core/container/src/main/java/org/wildfly/swarm/Swarm.java]: In this case environment variable must be called `swarm.project.stage`, what I would do is a `PROJECT_STAGE.replace('.', '_')` to be more environment fashion.

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java (hunk 1 above)
Comment (gastaldi) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java:467]: Thank you for your feedback, but I believe that the if/else approach is good in this case :+1:

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java (hunk 2 above)
Comment (gastaldi) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java]: I agree, I changed it to be `SWARM_PROJECT_STAGE`

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java (hunk 1 above)
Comment (kenfinnigan) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java:467]: What's the reason that Optional isn't a good approach here?

I'm a bit confused as I know @gastaldi has pushed the use of Optionals in other areas.

Comment (gastaldi) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java:467]: Not saying that Optional isn't a good approach, I just didn't want to spend much time on rewriting the whole method to use Optional (specially considering it does not belong in this Sprint 😜)

Comment (gastaldi) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java:467]: Also the suggested approach makes unnecessary calls if the property value exists, for example, so it should use orElseGet instead if converted

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**